From: Tom Gundersen <teg@jklm.no>
Date: Wed, 29 Jul 2015 01:34:35 +0200
Subject: networkd-wait-online: fix race

We must consider 'pending' links as if they may be managed by networkd, as this
is the state we enter before deciding wether networkd should manage the link
or not, so we better wait for this decision being made.
---
 src/network/networkd-wait-online-manager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/network/networkd-wait-online-manager.c b/src/network/networkd-wait-online-manager.c
index ec44893..9283f57 100644
--- a/src/network/networkd-wait-online-manager.c
+++ b/src/network/networkd-wait-online-manager.c
@@ -83,7 +83,7 @@ bool manager_all_configured(Manager *m) {
                         return false;
                 }
 
-                if (streq(l->state, "configuring")) {
+                if (STR_IN_SET(l->state, "configuring", "pending")) {
                         log_debug("link %s is being processed by networkd",
                                   l->ifname);
                         return false;
